public/assets/ts/admin/exam-bank.ts

```diff
@@ -0,0 +1,216 @@
+      } else {
+        table.classList.remove("filtered-code");
+      }
+      if (courseDeptSearchTokens.length > 0) {
+        table.classList.add("filtered-dept");
+      } else {
+        table.classList.remove("filtered-dept");
+      }
+    }
+
+    // apply filters
+    const examRows = Array.from(
+      table.querySelectorAll("tbody tr")
+    ) as HTMLTableRowElement[];
+
+    for (const examRow of examRows) {
+      const compareToken = (
+        tokenFromRow: string,
+        filterTokens: string[],
+        matchClass: string
+      ) => {
+        const departmentTokensMatchExam: boolean = filterTokens
+          .filter((token) => token.length > 0)
+          .reduce((foundAPreviousToken: boolean, currentToken: string) => {
+            return (
+              foundAPreviousToken ||
+              tokenFromRow.includes(currentToken.toLowerCase())
+            );
+          }, false);
+
+        if (departmentTokensMatchExam) {
+          examRow.classList.add(matchClass);
+        } else {
+          examRow.classList.remove(matchClass);
+        }
+      };
+
+      const examDepartment = examRow.getAttribute("data-course-dept");
+      compareToken(
+        examDepartment.toLowerCase(),
+        courseDeptSearchTokens,
+        "dept-match"
+      );
+
+      const examCourseCode = examRow.getAttribute("data-course-code");
+      compareToken(
+        examCourseCode.toLowerCase(),
+        courseCodeSearchTokens,
+        "code-match"
+      );
+    }
+  }
+
+  /**
+   * Builds the exam bank table
+   */
+  static async populateTable() {
+    const exams: Exam[] = await fetch("/api/exams").then((response) =>
+      response.json()
+    );
+
+    const tableBody = this.getExamList().querySelector("tbody");
+    const hiddenRow = tableBody.querySelector("#hidden-row");
+
+    for (const exam of exams) {
+      const newRow = hiddenRow.cloneNode(true) as HTMLElement;
+      newRow.id = "";
+      newRow.innerHTML = newRow.innerHTML
+        .replace("$COURSE-NAME$", `${exam.department} ${exam.courseCode}`)
+        .replace("$COURSE-DEPT$", exam.department)
+        .replace("$COURSE-CODE$", exam.courseCode)
+        .replace("$OFFERING$", exam.termName)
+        .replace("$TYPE$", exam.type)
+        .replace("$EXAM-FILE$", `/exams/${exam.examFile}`)
+        .replace("$SOLUTION-FILE$", `/exams/${exam.solutionFile}`)
+        .replace(
+          "$EXAM_HIDDEN$",
+          exam.examFile?.includes("-hidden") ? "True" : "False"
+        )
+        .replace(
+          "$SOLUTION_HIDDEN$",
+          exam.solutionFile?.includes("-hidden") ? "True" : "False"
+        );
+
+      newRow.setAttribute("data-course-dept", exam.department);
+      newRow.setAttribute("data-course-code", exam.courseCode);
+
+      const hideExamButton = newRow.getElementsByClassName("exam-actions")[0]
+        .children[1] as HTMLButtonElement;
+      const hideExamSolutionButton = newRow.getElementsByClassName(
```

Comment:
Can we prevent the solution buttons from displaying when there's no solution file?

Similarly, can we prevent the exam buttons from displaying when there's a solution file without an exam file?